lib/user_search_string.rb

```diff
@@ -0,0 +1,6 @@
+class UserSearchString < String
```

Comment:
True, but it's not as if this class prevents you from having search strings that aren't "valid": `UserSearchString.new("@foo")` will return something equal to `"foo"`, but `UserSearchString.new("foo").prepend("@")` will get you one that is unfiltered. Inheriting from String got you a load of methods that allow you to change the string "unsafely". Not just the ones that modify one in-place either. Just plain `UserSearchString.new("@foo").gsub(/^/, "@")` will return a *new* UserSearchString that still has the at-symbol in front, just because it doesn't call the initializer again.

I know nobody will probably try to do this. Until one morning when I haven't had coffee and I do. 

All you're really trying to do is converting a user-input to something for internal use. That's just a function that takes a string, and returns a string. Why complicate things with something that works only if you know all the pitfalls?